lib/java/spring/auth/src/main/java/co/airy/spring/auth/oidc/EmailFilter.java

```diff
@@ -0,0 +1,58 @@
+package co.airy.spring.auth.oidc;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class EmailFilter extends OncePerRequestFilter {
+    private final ConfigProvider configProvider;
+
+    public EmailFilter(ConfigProvider config) {
+        this.configProvider = config;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest req,
+                                    HttpServletResponse res,
+                                    FilterChain chain) throws IOException, ServletException {
+        final Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth instanceof OAuth2AuthenticationToken) {
```

Comment:
If the oauth flow is complete we still need to check if the authenticated user's email address matches the `allowedEmailPatterns` config